editor/utils/dom.js

```diff
@@ -56,18 +57,85 @@ export function isEdge( container, start = false ) {
 }
 
 /**
- * Places the caret at start or end of a given element
+ * Holds the first caret boundary rectangle for subsequent calls.
+ *
+ * @type {DOMRect}
+ */
+let firstVerticalRect;
```

Comment:
My biggest concern is this global here. It's a concern in general, but aggravated by the fact that we are in a `utils` module, where the expectation of lingering state is low, _and_ that the main function that sets it is a seemingly innocent predicate, `isVerticalEdge`.

I see it as a definite code smell that we should tackle, probably before merging. Does it mean that the predicate should become something else? That it should be broken into different pieces? That it should become a function returning a `DOMRect` or `null`? Relatedly, `onMouseDown={ resetVerticalPosition }` on WritingFlow is an implementation detail that shouldn't be leaking.

Reply:
Any ideas would we welcome! I'll have a think.

Reply:
Can firstVerticalRect just be passed in, and have its state / value managed in WritingFlow itself? It is essentially part of the state of writing flow, if that's where we are doing the cursor movement.

Reply:
Okay, I changed this.